itwm_example/csv_writer/csv_writer.py

```diff
@@ -1,29 +1,24 @@
-from traits.api import Instance
+from force_bdss.api import BaseCSVWriter, BaseCSVWriterFactory
 
-import csv
 
-from force_bdss.api import (
-    BaseNotificationListener,
-    MCOProgressEvent
-)
-from itwm_example.csv_writer.csv_writer_model import CSVWriterModel
+class CSVWriter(BaseCSVWriter):
+    def parse_progress_event(self, event):
+        event_datavalues = super().parse_progress_event(event)
+        event_datavalues.extend(event.weights)
+        return event_datavalues
```

Comment:
This should be implemented as a part of a custom `MCOProgressEvent`, i.e.
```
class CustomMCOProgressEvent(MCOProgressEvent):
    weights = List(Float())
    def serialize(self):
        return super().serialize() + self.weights
```

Reply:
In fact, this would also require the `MCO.run()` method to do something like `notify(CustomMCOProgressEvent(...))` instead of `notify_new_point(...)`.

Reply:
Yes, this feels like a more appropriate method to broadcast any `BaseDriverEvent` from the MCO. We should document this in an issue in the `force-bdss` repo and suggest a solution in a separate PR.